src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.NodesAndTokensToReduceComputer.cs

```diff
@@ -98,6 +101,14 @@ public override SyntaxNode Visit(SyntaxNode node)
 
             _simplifyAllDescendants = savedSimplifyAllDescendants;
             return node;
+
+            // While this is definitely not correct, enabling this behavior
+            // just for ForStatementSyntax will work correctly in some
+            // cases we care about without risking breaking anything.
```

Comment:
This smells like a bad design. You solution should be generic and if possible not depend on any concrete syntax node types. I doubt that even if you fix all other issues in this PR it can be accepted in such state

Reply:
Yeah, agreed